pandas/_libs/tslibs/conversion.pyx

```diff
@@ -285,11 +291,22 @@ cdef _TSObject convert_to_tsobject(object ts, tzinfo tz, str unit,
             obj.value = ts
             pandas_datetime_to_datetimestruct(ts, NPY_FR_ns, &obj.dts)
     elif PyDateTime_Check(ts):
-        return convert_datetime_to_tsobject(ts, tz, nanos)
+        if nanos == 0:
+            if isinstance(ts, ABCTimestamp):
+                reso = abbrev_to_npy_unit(ts.unit)  # TODO: faster way to do this?
+            else:
+                # TODO: what if user explicitly passes nanos=0?
```

Comment:
Is it possible to hit this? Maybe we should raise instead?

Reply:
i think it could happen with `pd.Timestamp(pydatetime_obj, nanosecond=0)`